common/lib/xmodule/xmodule/tabs.py

```diff
@@ -911,23 +309,16 @@ def validate_tabs(cls, tabs):
         if len(tabs) < 2:
             raise InvalidTabsException("Expected at least two tabs.  tabs: '{0}'".format(tabs))
 
-        if tabs[0].get('type') != CoursewareTab.type:
+        if tabs[0].get('type') != 'courseware':
             raise InvalidTabsException(
                 "Expected first tab to have type 'courseware'.  tabs: '{0}'".format(tabs))
 
-        if tabs[1].get('type') != CourseInfoTab.type:
+        if tabs[1].get('type') != 'course_info':
             raise InvalidTabsException(
                 "Expected second tab to have type 'course_info'.  tabs: '{0}'".format(tabs))
 
         # the following tabs should appear only once
-        for tab_type in [
-                CoursewareTab.type,
-                CourseInfoTab.type,
-                NotesTab.type,
-                TextbookTabs.type,
-                PDFTextbookTabs.type,
-                HtmlTextbookTabs.type,
-                EdxNotesTab.type]:
+        for tab_type in ['courseware', 'course_info', 'notes', 'textbooks', 'pdf_textbooks', 'html_textbooks']:
```

Comment:
In fact, all 'dynamic' tabs should only be included once, so maybe that is implied. The logic for dynamic tabs assumes that they can only be added once. Maybe instead we should have ``allow_multiple`` which defaults to ``False``, and only override it for the very few tabs that can be added multiple times (is it just static?).